blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/MaxValue.java

```diff
@@ -100,17 +100,17 @@ void handleRollupMetric(IBasicRollup basicRollup) throws RuntimeException {
         }
 
         if (this.isFloatingPoint() && !other.isFloatingPoint()) {
-            if (this.toDouble() < (double)other.toDouble()) {
+            if (this.toDouble() < (double)other.toLong()) {
```

Comment:
`MaxValue` can store a floating-point (`Double`) or a non-floating-point (`Long`) values. It does this by storing each in a separate field. If you call `toDouble()`, it returns the `double` field. If you call `toLong()`, it returns the `long` field. It is the job of the caller to determine which should be used, by calling `isFloatingPoint()`. [I don't like this system. That's just how the code behaves right now.]
In this particular case, it first checks that `this` holds a floating point number and that `other` holds a non-floating-point number. If that's the case, it should compare the respective `double` and `long` values. However, if `other` is not a floating-point number, and its `double` field has never been set, calling `toDouble()` here will return `0.0`, instead of the non-floating-point value which _was_ set.
In summary: `MaxValue` requires a two-step process for getting the correct stored value, and the original developers called the wrong method in step 2. Compare this to `MinValue`, which already used the correct `toLong()` at this point: https://github.com/izrik/blueflood/blob/test-maxvalue/blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/MinValue.java#L103